meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -414,4 +415,128 @@ private static EntryMode convertZkEntryMode(long ephemeralOwner) {
         throw new IllegalArgumentException(zkEphemeralType + " is not supported.");
     }
   }
+
+
+  @Override
+  public List<OpResult> transactionOP(Iterable<Op> iterable) {
+    // Convert list of MetaClient Ops to Zk Ops
+    List<org.apache.zookeeper.Op> zkOps = metaClientOpToZk(iterable);
+    // Execute Zk transactional support
+    List<org.apache.zookeeper.OpResult> zkResult = _zkClient.multi(zkOps);
+    // Convert list of Zk OpResults to MetaClient OpResults
+    return zkOpResultToMetaClient(zkResult);
+  }
+
+  /**
+   * Helper function for transactionOp. Converts MetaClient Op's into Zk Ops to execute
+   * zk transactional support.
+   * @param ops
+   * @return
+   */
+  public List<org.apache.zookeeper.Op> metaClientOpToZk(Iterable<Op> ops) {
+    List<org.apache.zookeeper.Op> zkOps = new ArrayList<>();
+    org.apache.zookeeper.Op temp;
+    for (Op op : ops) {
+      switch (op.getType()) {
+        case CREATE:
+          int zkFlag = zkFlagFromEntryMode(((Op.Create) op).getEntryMode());
+          try {
+            temp = org.apache.zookeeper.Op.create(
+                op.getPath(), ((Op.Create) op).getData(), DEFAULT_ACL, CreateMode.fromFlag(zkFlag));
+          } catch (KeeperException e) {
+            throw new MetaClientKeeperException(e);
+          }
+          break;
+        case DELETE:
+          temp = org.apache.zookeeper.Op.delete(
+              op.getPath(), ((Op.Delete) op).getVersion());
+          break;
+        case SET:
+          temp = org.apache.zookeeper.Op.setData(
+              op.getPath(), ((Op.Set) op).getData(), ((Op.Set) op).getVersion());
+          break;
+        case CHECK:
+          temp = org.apache.zookeeper.Op.check(
+              op.getPath(), ((Op.Check) op).getVersion());
+          break;
+        default:
+          throw new IllegalArgumentException(op.getType() + " is not supported.");
+      }
+      zkOps.add(temp);
+    }
+    return zkOps;
+  }
+
+  /**
+   * Helper function for transactionOP. Converts the result from calling zk transactional support into
+   * metaclient OpResults.
+   * @param zkResult
+   * @return
+   */
+  public List<OpResult> zkOpResultToMetaClient(List<org.apache.zookeeper.OpResult> zkResult) {
+    List<OpResult> metaClientOpResult = new ArrayList<>();
+    OpResult temp;
+    String resultClass;
+    for (org.apache.zookeeper.OpResult opResult : zkResult) {
+      Stat metaClientStat;
+      resultClass = opResult.getClass().getSimpleName();
+      switch (resultClass) {
+        case "CreateResult":
```

Comment:
Comparing on classname is too risky and low on performance (it's still magic number to some extent). If you must do it this way, please check `instance of`.

Reply:
I see the difficulties here in binding two classes without knowing the type in compile time. One potential way to improve similar code, is to try using a static map to store the binding and utilize lambda functional.
e.g.
```
// mapping from class object => construction function of OpResult
Map<Class<? extends org.apache.zookeeper.OpResult>, Function<org.apache.zookeeper.OpResult, OpResult>> map = new HashMap<>();
map.put(org.apache.zookeeper.OpResult.CheckResult.class, op -> new OpResult.CheckResult()); 
for (org.apache.zookeeper.OpResult opResult : zkResult) {
  map.get(opResult.getClass()).apply(opResult);
}

```